src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -590,6 +592,7 @@ class LeonardoService(protected val dataprocConfig: DataprocConfig,
                gdDAO.resizeCluster(cluster.googleProject, cluster.clusterName, numPreemptibles = Some(0))
              else Future.unit
 
+        _ <- welderDao.flushCache(cluster.googleProject, cluster.clusterName).handleError(e => logger.error(s"fail to flush welder cache for ${cluster}"))
```

Comment:
Does the flush API call block until the cache is written to disk? Just thinking about race conditions: we don't want to stop the instance before the cache is actually flushed.

Reply:
yep. it blocks

Reply:
Also: should we only do this if welder is enabled?